packages/core/src/common/messaging/proxy-factory.ts

```diff
@@ -143,17 +143,18 @@ export class JsonRpcProxyFactory<T extends object> implements ProxyHandler<T> {
      *
      * @returns A promise of the method call completion.
      */
-    protected onRequest(method: string, ...args: any[]): Promise<any> {
-        return new Promise<any>((resolve, reject) => {
-            try {
-                const promise = this.target[method](...args) as Promise<any>;
-                promise
-                    .catch(err => reject(err))
-                    .then(result => resolve(result));
-            } catch (err) {
-                reject(err);
+    protected async onRequest(method: string, ...args: any[]): Promise<any> {
```

Comment:
Cool! I am glad you have fixed this. We will receive real errors on the frontend with this change, right?

Reply:
yes